syntax = "proto3";

package memoryservice;

// Opciones para mejorar compatibilidad con compiladores y sistemas
option cc_enable_arenas = true;
option go_package = "memoryservicepb";

service MemoryService {
  rpc Create (CreateRequest) returns (CreateResponse);
  rpc Set (SetRequest) returns (SetResponse);
  rpc Get (GetRequest) returns (GetResponse);
  rpc IncreaseRef (RefRequest) returns (RefResponse);
  rpc DecreaseRef (RefRequest) returns (RefResponse);
}

message CreateRequest {
  int32 size = 1;
  string type = 2;
}

message CreateResponse {
  int32 id = 1;
}

message SetRequest {
  int32 id = 1;
  string value = 2;
}

message SetResponse {
  bool success = 1;
}

message GetRequest {
  int32 id = 1;
}

message GetResponse {
  string value = 1;
}

message RefRequest {
  int32 id = 1;
}

message RefResponse {
  bool success = 1;
}

